src/main/java/Duke.java

```diff
@@ -1,10 +1,378 @@
+
+        String line = "---------------------------------------------";
+        String greet = "Hello! I'm Duke";
+        String question = "What can I do for you?";
+        System.out.println(line + "\n" + greet + "\n" + question + "\n" + line);
+
+        boolean run = true;
+        String[] tasks = new String[100];
+        String[] marks = new String[100];
+        String[] type = new String[100];
+        int count = 1;
+
+        while(run){
+            Scanner in = new Scanner(System.in);
+            String text;
+            text = in.nextLine();
+            if(text.equals("bye")){
+                System.out.println(line);
+                System.out.println("Bye. Hope to see you again soon!");
+                System.out.println(line);
+                run = false;
+            }
+            else if (text.equals("list")){
+                System.out.println(line);
+                System.out.println("Here are the tasks in your list:");
+                for(int i = 1; i< count; i++){
+                    System.out.println(i + ". " + type[i] + marks[i] + " " + tasks[i]);
+                }
+                System.out.println(line);
+            }
+
+            else if (text.contains("mark")){
+                System.out.println(line);
+                String strNum;
+                int num;
+                if(text.contains("unmark")){ //Unmark
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    strNum = text.substring(7, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[ ]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+
+                }
+                else{ //Mark
+                    System.out.println("Nice! I've marked this task as done:");
+                    strNum = text.substring(5, text.length());
+                    num = Integer.parseInt(strNum);
+                    marks[num] = "[X]";
+                    System.out.println(marks[num] + " " + tasks[num]);
+                }
+
+                System.out.println(line);
+            }
+
+            else if (text.contains("todo")) {
+                if(text.equals("todo") || text.equals("todo ")){
+                    System.out.println(line);
+                    System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println(line);
+                } else{
+                    System.out.println(line);
+                    System.out.println("Got it. I've added this task");
+                    text = text.substring(5, text.length());
+                    System.out.println("  [T][ ] " + text);
+                    System.out.println("Now you have " + count + " tasks in the list.");
+                    System.out.println(line);
+                    tasks[count] = text;
+                    type[count] = "[T]";
+                    marks[count] = "[ ]";
```

Comment:
Avoid magic strings. Consider puttiing the strings into a descriptive variable. For example, `[T]` can be added to the variable `TASK_TYPE_MESSAGE` or more appropriate names.